### PR TITLE
release 0.3.0

### PR DESCRIPTION
**Pull request by anthony-near**
Changes:
1. Consolidate key files into single json. Suggestion by @anshal-savla to make key management easier (1 credentials file easier to work with and from secretsmanager than multiple)
- [x] consolidate keys from multiple json files to single json file containing a list of json credentials
- [x] test locally
2. Fix to only check basic whitelist receiver_id if not fastauth. Fixes issues fastauth was experiencing due to changes made for other configurations of relayer. 
3. Tracing with support for flamechart and flamegraph generation from tracing output. To help identify performance bottlenecks, such as the [one identified](https://github.com/near/pagoda-relayer-rs/issues/40) with `/create_account_atomic` when deployed, more tracing features have been added. We can now see the time taken for each request to send a response:
```
2023-11-15T21:15:30.881115Z DEBUG request{method=GET uri=/get_allowance version=HTTP/1.1}: tower_http::trace::on_response: finished processing request latency=27 ms status=200
2023-11-15T21:16:00.230494Z DEBUG request{method=POST uri=/send_meta_tx version=HTTP/1.1}: tower_http::trace::on_response: finished processing request latency=3078 ms status=500
2023-11-15T21:16:32.513769Z DEBUG request{method=POST uri=/create_account_atomic version=HTTP/1.1}: tower_http::trace::on_response: finished processing request latency=2235 ms status=500
```

- [x]  add tracing `#[instrument]` attribute to each endpoint function and `process_signed_delegate_action` which enables the request latency in the tracing logs
- [x] support for flamegraph and flamechart generation using the https://crates.io/crates/tracing-flame crate
- [x] updated README with performance testing instructions
- [x] Reduces calls to clone and some other superfluous function calls or allocations.

- [x] updated near-fetch version and shared storage logic
- [x] feature flags for fastauth and shared storage
- [x] `process_signed_delegate_action_noretry_async` fn used by new endpoints `/send_meta_tx_async`, `/send_meta_tx_nopoll` - for high throughput transactions
- [x] use blocking, very slow retry only for create-account transactions for fastauth

Individuals PRs:
https://github.com/near/pagoda-relayer-rs/pull/44
https://github.com/near/pagoda-relayer-rs/pull/46
https://github.com/near/pagoda-relayer-rs/pull/47
https://github.com/near/pagoda-relayer-rs/pull/53
https://github.com/near/pagoda-relayer-rs/pull/57

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,12 +12,11 @@ target/
...
+
 # where account and keys are stored
 *.json
 # except for example json file used for testing
 !nomnomnom.testnet.json
-!37cb5b936de257d014e663663b186e58e4c3c4c9b4c7643e7b89821e789b0b4b.json
-!b278b7752b844a0cc88f8dbdb053b860d6283fb353e7a317b3b82efe01396c3e.json
-!df86e9701801e30cf0649b2ea43aeacef60e394e076709d4519880d5a9c7bc9e.json
-!e4a6dd159d45d4104679e709d29d49a533e5df9467b311c9dbea7fb8f0077335.json
-!d7adc371186bc2013ec5de36be27ed9a36022af1112f7c62530baf6042690f61.json
```
Comment (encody) [inline .gitignore:23]: Just double-checking: are these key files supposed to be included in the repo? They contain private (testnet) keys. It appears some of them may be duplicated in `account_keys/nomnomnom.testnet.json`.

File: 37cb5b936de257d014e663663b186e58e4c3c4c9b4c7643e7b89821e789b0b4b.json
Comment (encody) [inline 37cb5b936de257d014e663663b186e58e4c3c4c9b4c7643e7b89821e789b0b4b.json:1]: e.g. this file is included in the repo now.

File: .gitignore (hunk 1 above)
Comment (anthony-near) [reply .gitignore:23]: They are supposed to be included in the repo as they are all testnet keys. The purpose is to explicitly show the format necessary for others and to make it easy for running out of the box.

Comment (anthony-near) [reply .gitignore:23]: They are now moved into the nomnomnom.testnet.json file as one of the refactors done in this release is moving away from multiple individual key files, which can be hard to manage for deployments, to a single list of keys in a single file.

**Review by encody — APPROVED** [review]

**Comment by firatNEAR** [comment]
What is the release schedule for this release? Are we looking to get it in after the holiday season?

**Comment by anthony-near** [comment]
> What is the release schedule for this release? Are we looking to get it in after the holiday season?

I want to merge this into main next week after I finish this PR https://github.com/near/pagoda-relayer-rs/pull/57. The fastauth team said that they will test it in their own time.

**Pull request merged**